test/e2e/report_dynamic_input_data_test.go

```diff
@@ -163,3 +169,129 @@ func testReportsProduceData(t *testing.T, testReportingFramework *reportingframe
+			Expression: "*/1 * * * *",
+		},
+	}
+	var foundOnce bool
+	expiration := &metav1.Duration{Duration: 30.0 * time.Second}
+	// If a report is in dependency unmet state it won't be deleted and it can take awhile for import to happen.
+	// 4 minutes seems to be enough time for that setup, and the delete, to reliably happen.
+	waitTimeWhileCheckDeletion := 4 * time.Minute
+
+	report := testReportingFramework.NewSimpleReport(
+		"report-should-delete",
+		"namespace-memory-request",
+		cronSchedule,
+		nil,
+		nil,
+		expiration,
+	)
+	t.Logf("creating report %s and waiting %s to finish", report.Name, waitTimeWhileCheckDeletion)
+	createErr := testReportingFramework.CreateMeteringReport(report)
+	if createErr != nil {
+		assert.Failf(t, "Creating report should not have an error", createErr.Error())
+	}
+	err := wait.Poll(time.Second*5, waitTimeWhileCheckDeletion, func() (bool, error) {
+		_, err := testReportingFramework.GetMeteringReport("report-should-delete")
+		// `foundOnce` provides a latch that allows us to wait for report creation then exit on deletion
+		if err == nil {
+			foundOnce = true
+		}
+		if err != nil {
+			if foundOnce && strings.Contains(err.Error(), "not found") {
+				return true, errors.New("report was deleted after being created")
+			}
+		}
+		// if we're exiting due to timeout, it's a fail
+		return false, nil
+	})
+	if err != nil {
+		assert.Contains(t, err.Error(), "was deleted", "report should have been deleted by retention period being reached")
+	}
+}
+
+func testReportIsNotDeletedWhenReportDependsOnIt(t *testing.T, testReportingFramework *reportingframework.ReportingFramework) {
+	testQueryName := "namespace-cpu-usage"
+
+	// cron schedule to run every minute
+	cronSchedule := &metering.ReportSchedule{
+		Period: metering.ReportPeriodCron,
+		Cron: &metering.ReportScheduleCron{
+			Expression: "*/1 * * * *",
+		},
+	}
+	var foundOnce = false
+	expiration := &metav1.Duration{Duration: 30.0 * time.Second}
+	// If a subReport is in dependency unmet state it won't be deleted and it can take awhile for import to happen.
+	// 4 minutes seems to be enough time for that setup, and the delete, to reliably happen.
+	waitTimeWhileCheckDeletion := 4 * time.Minute
+
+	subReport := testReportingFramework.NewSimpleReport(
+		"subreport-should-not-delete",
+		testQueryName,
+		cronSchedule,
+		nil,
+		nil,
+		expiration,
+	)
+	t.Logf("creating subReport %s and waiting %s to finish", subReport.Name, waitTimeWhileCheckDeletion)
+
+	createErr := testReportingFramework.CreateMeteringReport(subReport)
+	if createErr != nil {
+		assert.Failf(t, "Creating subReport should not have an error", createErr.Error())
+	}
+
+	newDefault := func(s string) *json.RawMessage {
+		v := json.RawMessage(s)
+		return &v
+	}
+
+	report := testhelpers.NewReport(
+		"report-depends-on-subreport",
+		testReportingFramework.Namespace,
+		testQueryName,
+		[]metering.ReportQueryInputValue{
+			metering.ReportQueryInputValue{
+				Name: "Report",
```

Comment:
This is [not a valid input](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/kube-reporting_metering-operator/1367/pull-ci-kube-reporting-metering-operator-master-metering-e2e-aws/1301934731100688384/artifacts/metering-e2e-aws/metering_test_output/e2e/ReportExpiration/logs/namespaces/metering-reportexpiration/metering.openshift.io/reports/report-depends-on-subreport.yaml).

Reply:
I tried swapping out a valid input, `NamespaceCPUUsageReportName` and that yields a valid report, so how would I programmatically link the sub Report to that input then is what I am still wondering.

Reply:
That seems sensible